plugins/system/mvcoverride/helper/mvcloader.php

```diff
@@ -0,0 +1,239 @@
+	 * @param   string  $suffix          Suffix for extend files
+	 *
+	 * @return void
+	 */
+	public static function setOverrideFile($class, $path, $isOverrideFile = false, $prefix = null, $suffix = null)
+	{
+		$object = new stdClass;
+		$object->path = $path;
+		$object->isOverride = $isOverrideFile;
+		$object->prefix = $prefix;
+		$object->suffix = $suffix;
+
+		if ($isOverrideFile)
+		{
+			$class = $prefix . $class . $suffix;
+		}
+
+		// Sanitize class name.
+		$class = strtolower($class);
+		self::$overrideFiles[$class] = $object;
+	}
+
+	/**
+	 * Method to setup the autoloaders for the Joomla Platform.
+	 *
+	 * @param   int     $changePrivate  Flag for change values and functions to protected instead private in extends files
+	 * @param   string  $prefix         Prefix for extend files
+	 * @param   string  $suffix         Suffix for extend files
+	 *
+	 * @return  void
+	 */
+	public static function setupOverrideLoader($changePrivate = 0, $prefix = '', $suffix = 'Default')
+	{
+		self::$changePrivate = $changePrivate;
+		self::$prefix = $prefix;
+		self::$suffix = $suffix;
+
+		// Register the prefix autoloader.
+		spl_autoload_register(array('MVCLoader', '_load_override'), false, true);
+	}
+
+	/**
+	 * Autoload function for override files
+	 *
+	 * @param   string  $class  Name class search
+	 *
+	 * @return bool
+	 */
+	public static function _load_override($class)
+	{
+		// Sanitize class name.
+		$lowerClass = strtolower($class);
+
+		// If the class already exists do nothing.
+		if (class_exists($lowerClass, false))
+		{
+			return true;
+		}
+
+		// If the class is registered include the file.
+		if (isset(self::$classes[$lowerClass]))
+		{
+			if (self::checkOverride(self::$classes[$lowerClass], $lowerClass))
+			{
+				return true;
+			}
+		}
+
+		foreach (self::$prefixes as $prefix => $lookup)
+		{
+			$chr = strlen($prefix) < strlen($class) ? $class[strlen($prefix)] : 0;
+
+			if (strpos($class, $prefix) === 0 && ($chr === strtoupper($chr)))
+			{
+				if (self::_load(substr($class, strlen($prefix)), $lookup, $prefix))
+				{
+					return true;
+				}
+			}
+		}
+
+		if (isset(self::$overrideFiles[$lowerClass]))
+		{
+			$currentData = self::$overrideFiles[$lowerClass];
+
+			if (isset($currentData->isOverride) && $currentData->isOverride)
+			{
+				if (self::loadOverrideFile($currentData->path, $currentData->prefix, $currentData->suffix))
+				{
+					return true;
+				}
+			}
+			else
+			{
+				return include $currentData->path;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Load a class based on name and lookup array.
+	 *
+	 * @param   string  $class   The class to be loaded (wihtout prefix).
+	 * @param   array   $lookup  The array of base paths to use for finding the class file.
+	 * @param   string  $prefix  Prefix part extension
+	 *
+	 * @return  boolean  True if the class was loaded, false otherwise.
+	 */
+	private static function _load($class, $lookup, $prefix)
```

Comment:
Not an issue but: not sure why you start with underscore the method. If you do due to the fact that is private you are not doing so at line 102 (public method starting with underscore). Also we don't use underscore but CamelCased according to coding standards https://redweb.atlassian.net/wiki/display/Devportal/Variable+names+and+other+naming+conventions#Variablenamesandothernamingconventions-UseCamelCasenotunderscores